api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+swagger: '2.0'
+info:
+  title: Nakadi Event Bus API Definition
+  description: |
+    -------------------------------
+    Nakadi Event Bus API Definition
+    -------------------------------
+
+    * Low-level API
+    * Stantardised event format (see #/definitions/Event)
+    * Partition strategy: event store will infer the partition from the `partitioning_key` in the message body
+
+  version: '0.4'
+  contact:
+    name: Team Aruha @ Zalando
+    email: nakadi-users@zalando.de
+schemes:
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  oauth2:
+    type: oauth2
+    flow: implicit
+    authorizationUrl: 'https://auth.example.com/oauth2/tokeninfo'
+
+paths:
+  /metrics:
+    get:
+      tags:
+        - monitoring
+      summary: Get monitoring metrics
+      responses:
+        '200':
+          description: Metrics data
+          schema:
+            $ref: '#/definitions/Metrics'
+        '401':
+          description: User is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Not available
+          schema:
+            $ref: '#/definitions/Problem'
+  /topics:
+    get:
+      description: Lists all known `Topics` of this Nakadi cluster
+      tags:
+        - monitoring
+      responses:
+        '200':
+          description: The request was successful. Returns a list `Topic`s in the response body.
+          schema:
+            type: array
+            description: An array of topics
+            items:
+              $ref: '#/definitions/Topic'
+        '401':
+          description: User is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '503':
+          description: Not available
+          schema:
+            $ref: '#/definitions/Problem'
+  '/topics/{topic}/events':
+    get:
+      tags:
+        - low-level-api
+      summary: |
+        Starts a stream delivery for the specified partitions of the given topic.
+        The tracking of the current position in the partitions and of which
+        partitions is being read is in the responsibility of the client.
+        No commits are needed.
+      parameters:
+        - name: topic
+          in: path
+          description: Topic where to get events from
+          type: string
+          required: true
+        - name: x-nakadi-cursors
+          in: header
+          description: Cursors pointing to partitions/offsets to start read from
+          required: false
+          type: array
```

Comment:
I'm not sure whether Swagger supports JSON in headers. Arrays in headers only support primitive elements. See https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#headerObject.

> **Required.** The internal type of the array. The value MUST be one of `"string"`, `"number"`, `"integer"`, `"boolean"`, or `"array"`. Files and models are not allowed.